.github/workflows/release.yml

```diff
@@ -1,11 +1,53 @@
-name: Publish Release
+name: Test and publish a new release
   
 on:
-  release:
-    types: [published]
-    branches: [master]
+  push:
+    tags:
+      - 'v*'
```

Comment:
shouldn't there be restriction to branch name too? like what will happen, if tag is pushed on non-master branch?

Reply:
i assume it would checkout, build and release that tag. do you maybe know if that's the case? if so then it's even better at first glance since we could build older CE2 branches if it's necessary for some reason

Reply:
I believe that there is a way to allow workflow trigger for only limited set of branches, something like:
```yaml
    branches: [master, series-0.x]
```

Reply:
not sure it's possible without some workarounds due to the nature of tags, see https://github.com/orgs/community/discussions/13226.  
it gets even more complicated if we move it to a generic action in `evolution-gaming/scala-github-actions` since different repos have their own legacy branches with different names so they'd have to work around that

Reply:
we can pass arguments to action

Reply:
yes, though i'm not convinced yet it's a good tradeoff to implement quite a workaround (see [this stackoverflow answer](https://stackoverflow.com/questions/63745613/how-to-get-a-branch-name-on-github-action-when-push-on-a-tag/77727776#77727776)) to just ensure the tag exists on a specific branch

what are the exact benefits we're supposed to receive by introducing this complexity?

Reply:
not sure, but technically we, as Software Engineers, can try to do that 😄